sormas-backend/src/main/java/de/symeda/sormas/backend/person/PersonFacadeEjb.java

```diff
@@ -954,26 +962,66 @@ public void onPersonChanged(PersonDto existingPerson, Person newPerson) {
 
 		// Update cases if present condition has changed
 		if (existingPerson != null) {
+			// sort cases based on recency
+			Collections.sort(
+				personCases,
+				(c1, c2) -> CaseLogic.getStartDate(c1.getSymptoms().getOnsetDate(), c1.getReportDate())
+					.before(CaseLogic.getStartDate(c2.getSymptoms().getOnsetDate(), c2.getReportDate())) ? 1 : -1);
+
 			if (newPerson.getPresentCondition() != null && existingPerson.getPresentCondition() != newPerson.getPresentCondition()) {
 				// Update case list after previous onCaseChanged
-				personCases = caseService.findBy(new CaseCriteria().person(new PersonReferenceDto(newPerson.getUuid())), true);
-				for (Case personCase : personCases) {
-					if (newPerson.getPresentCondition().isDeceased()) {
-						if (personCase.getOutcome() == CaseOutcome.NO_OUTCOME) {
-							CaseDataDto existingCase = CaseFacadeEjbLocal.toDto(personCase);
-							personCase.setOutcome(CaseOutcome.DECEASED);
-							personCase.setOutcomeDate(newPerson.getDeathDate() != null ? newPerson.getDeathDate() : new Date());
-							// Attention: this may lead to infinite recursion when not properly implemented
-							caseFacade.onCaseChanged(existingCase, personCase);
-						}
-					} else if (personCase.getOutcome() == CaseOutcome.DECEASED) {
+				Case personCase = personCases.isEmpty() ? null : personCases.get(0);
+				if (newPerson.getPresentCondition().isDeceased()
+					&& newPerson.getDeathDate() != null
+					&& newPerson.getCauseOfDeath() == CauseOfDeath.EPIDEMIC_DISEASE
+					&& newPerson.getCauseOfDeathDisease() != null) {
+
+					// update the latest associated case
+					if (personCase != null
+						&& personCase.getOutcome() != CaseOutcome.DECEASED
+						&& (personCase.getReportDate().before(DateHelper.addDays(newPerson.getDeathDate(), 30))
+							&& personCase.getReportDate().after(DateHelper.subtractDays(newPerson.getDeathDate(), 30)))) {
+						CaseDataDto existingCase = CaseFacadeEjbLocal.toDto(personCase);
+						personCase.setOutcome(CaseOutcome.DECEASED);
+						personCase.setOutcomeDate(newPerson.getDeathDate());
+						caseFacade.onCaseChanged(existingCase, personCase);
+					}
+				} else if (!newPerson.getPresentCondition().isDeceased()
+					&& (existingPerson.getPresentCondition() == PresentCondition.DEAD
+						|| existingPerson.getPresentCondition() == PresentCondition.BURIED)) {
+					// Person was put "back alive"
+					// make sure other values are set to null
+					newPerson.setCauseOfDeath(null);
+					newPerson.setCauseOfDeathDisease(null);
+					newPerson.setDeathPlaceDescription(null);
+					newPerson.setDeathPlaceType(null);
+					newPerson.setBurialDate(null);
+					newPerson.setCauseOfDeathDisease(null);
+					// update the latest associated case, if it was set to deceased && and if the case-disease was also the causeofdeath-disease
+					if (personCase != null
```

Comment:
I think this as well should probably always take the latest case of the specified disease that is within the threshold, not strictly the latest case if it is of a different disease

Reply:
So should I just filter the `personCases` List based on `disease==causeOfDeathDisease` before doing any of the other logic?